src/PerpEF.sol

```diff
@@ -223,6 +218,12 @@ contract PerpEF is ERC20, Ownable {
             sizeInIndexTokens: sizeInIndexTokens
         });
 
+        // Collateral transfer
+        if (
+            i_collateralToken.allowance(msg.sender, address(this)) < collateral
+        ) {
+            revert PerpEF__NotEnoughAllowance();
+        }
```

Comment:
This allowance check is better at the beginning of the function, because if this check fails, the user won't loose funds for unnecessary gas usage. This is a gas optimization best practice.